src/library/scala/collection/Iterator.scala

```diff
@@ -1259,7 +1259,7 @@ object Iterator extends IterableFactory[Iterator] {
         else adjustedBound min (until - lo)   // keep lesser bound
       if (rest == 0) empty
       else {
-        dropping += lo
+        dropping = math.min(Int.MaxValue.toLong, lo.toLong + dropping.toLong).toInt
```

Comment:
```scala
val sum = lo + dropping 
if (sum < 0) Int.MaxValue else sum
``` 
how about this?

Reply:
I don't see the difference

Reply:
the math.min is a method call

Reply:
The Long conversion is required I think. Simply checking whether the int has overflowed might not work given something like this

```
.drop(IntMax).drop(IntMax) and so on
```
 Because that would turn the variable back to positive though unlikely

Reply:
Have you tested it? I think it will not reach the positive again.
```scala
        dropping = {
          val sum = dropping + lo
          if (sum < 0) Int.MaxValue else sum
        }
```

Reply:
Ah yes missed the part where we set it to maxvalue. Yeah that works

Reply:
Anyways, updated as mentioned